src/main/kotlin/no/elhub/auth/features/documents/create/PdfFactory.kt

```diff
@@ -0,0 +1,104 @@
+package no.elhub.auth.features.documents.create
+
+import arrow.core.Either
+import arrow.core.getOrElse
+import arrow.core.left
+import arrow.core.raise.either
+import com.github.mustachejava.DefaultMustacheFactory
+import com.openhtmltopdf.pdfboxout.PdfRendererBuilder
+import org.apache.pdfbox.Loader
+import org.apache.pdfbox.pdmodel.PDDocumentInformation
+import java.io.ByteArrayOutputStream
+import java.io.StringWriter
+
+typealias MustacheVariableTags = Map<String, String>
+
+typealias PdfBytes = ByteArray
+
+sealed interface PdfError {
+    data object HtmlGenerationError : PdfError
+    data object PdfGenerationError : PdfError
+    data object MetaDataError : PdfError
+}
+
+fun interface PdfFactory {
```

Comment:
:art: PDF is a concrete implementation, we should try to remain agnostic to the type of document we generate in the domain layer